tools/pybind11Tools.cmake

```diff
@@ -224,6 +224,14 @@ function(pybind11_add_module target_name)
     endif()
   endif()
 
+  # Python 2 doesn't really support C++17, Clang warns/errors over it
+  if(CMAKE_CXX_STANDARD
+     AND CMAKE_CXX_COMPILER_ID MATCHES "Clang"
+     AND PYTHON_VERSION VERSION_LESS 3.0
+     AND NOT CMAKE_CXX_STANDARD VERSION_LESS 17)
```

Comment:
I tend to avoid `VERSION_*_EQUAL` since they were added in CMake 3.7 (and I haven't broken old CMake support quite yet). But, you are right, this is just a number, so `GREATER_EQUAL` would be just fine.

Reply:
To be honest, all my rambling about special members can be ignored, but I think that simpler cmake is important. Cmake is horrible to read as it is.

Reply:
Fully agree about CMake, it should be as clean as possible. I'll be revamping it soon to drop 3.6 and below, which will help.

> Cmake is horrible to read as it is

CMake 2.8 is ugly. Modern CMake is not. If we dumped all compatibility with older versions and forced everyone to use the latests tools like `find_package(Python)`, we could have a fairly beautiful CMakeLists. But we can't do that - but maybe we can still improve things a lot.